src/main/java/carriermanagementsystem/CarrierManagementSystemDirector.java

```diff
@@ -19,48 +19,47 @@
  * carrier it creates.
  */
 public class CarrierManagementSystemDirector implements CarrierManagementIf {
-    private static volatile CarrierManagementSystemDirector UNIQUE_INSTANCE = new CarrierManagementSystemDirector();
-    private final int LTE_MAXIMUM_CARRIER_ID = 500; 
+    private static final CarrierManagementSystemDirector UNIQUE_INSTANCE = new CarrierManagementSystemDirector();
```

Comment:
Actually, this should still be volatile (at least according to what Ista deems as the only "correct" Singleton). And the change should be in the getInstance() method.

Reply:
I've been trying to figure out how volatile vs synchronized should be used and I saw most times it was either one or the other, but I'm not sure enough of that so I'll just put it back :)